pennylane/templates/state_preparations/basis.py

```diff
@@ -109,6 +110,13 @@ def compute_decomposition(basis_state, wires):  # pylint: disable=arguments-diff
         [PauliX(wires=['a']),
         PauliX(wires=['b'])]
         """
+        if len(qml.math.shape(basis_state)) > 1:
```

Comment:
why not qml.math.ndim? same below. tensorflow?

Reply:
I used `ndim` originally, remembered the tf discussion, and then changed it to `len(qml.math.shape)` :)